### PR TITLE
feat: install from package url

### PR DESCRIPTION
**Pull request by hypercall88**
# feat: install clamav from package url

## 🗣 Description ##

This change adds the feature to install from a package URL instead of through the OS distribution package manger. The following default variables are added:

| Variable | Description | Default | Required |
|----------|-------------|---------|----------|
| clamav_install_from_package_manager | A boolean value to determine if the role should install from the system package manager. | `true` | No |
| clamav_package_version | The package version to install from the URL if not installing from the system package manager. | `1.3.1` | No |

In addition, a root-owned freshclam systemd service in `/etc/systemd/system` and freshclam configuration file at the specified `freshclam_configuration_path` are created via template files since the ClamAV daemon package no longer handles this. 

The `cron` packages are added to their own `dependencies` list to separate from the ClamAV-specific packages to install.

A molecule test, `package_install`, was added to test this new feature.

## 💭 Motivation and context ##

This feature allows playbook users to install the ClamAV package with bundled dependencies directly from the [ClamAV download URL](https://www.clamav.net/downloads).

This change is required on [RHEL version prior to 8.6](https://access.redhat.com/solutions/6981276) due to the epel repo no longer providing the required libjson-c.so.4 library. Rhel 8.4 is still a critical version in cloud Azure subscriptions.

## 🧪 Testing ##

I ran the pre-commit scripts and added a molecule test for the new feature. I ran all the molecule tests.

~I also updated the ansible and molecule versions in the test requirements, as it seems newer versions of molecule solved the bug mentioned in molecule:~
```
# See also cisagov/skeleton-packer#312.
ansible-core<2.16.3
# With the release of molecule v5 there were some breaking changes so
# we need to pin at v5 or newer. However, v5.0.0 had an internal
# dependency issue so we must use the bugfix release as the actual
# lower bound.
```

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated to reflect the changes in this PR.
- [x] Tests have been added and/or modified to cover the changes in this PR.
- [x] All new and existing tests pass.

### CODE REVIEW

**Comment by jsf9k** [comment]
I will rebase this PR once #79 is merged.

**Comment by jsf9k** [comment]
> I will rebase this PR once #79 is merged.

@desebjohnston - I rebased, but I would need to force-push the changes to your branch.  Can I do that?  I did remove the changes you made to `requirements-test.txt` and `.pre-commit-config.yaml` since (1) I don't think they are necessary now that #79 has been merged and (2) those changes would have to be made to [cisagov/skeleton-ansible-role](https://github.com/cisagov/skeleton-ansible-role) and then allowed to flow down to that repo's descendants via [cisagov/action-lineage](https://github.com/cisagov/action-lineage).

**Comment by hypercall88** [comment]
> > I will rebase this PR once #79 is merged.
> 
> @desebjohnston - I rebased, but I would need to force-push the changes to your branch. Can I do that? I did remove the changes you made to `requirements-test.txt` and `.pre-commit-config.yaml` since (1) I don't think they are necessary now that #79 has been merged and (2) those changes would have to be made to [cisagov/skeleton-ansible-role](https://github.com/cisagov/skeleton-ansible-role) and then allowed to flow down to that repo's descendants via [cisagov/action-lineage](https://github.com/cisagov/action-lineage).

Yes, that's absolutely fine. Thanks for consuming my changes!

**Comment by jsf9k** [comment]
> Thanks for consuming my changes!

Thank _you_ for the contribution!

**Comment by hypercall88** [comment]
FYSA I have a few fixes I had to push after inspecting my logs and realizing the scan cronjob was failing this morning due to pathing issues.

I also fixed a couple linting issues.

**Comment by jsf9k** [comment]
@desebjohnston - The `modify_config` checks are failing because GH Actions has been hitting `freshclam` too hard and some of the IPs it uses have been blocked.  I'll handle those failures so you can focus on any failing `package_install` checks that are actually relevant to this PR.

**Comment by jsf9k** [comment]
The remaining builds that are failing are doing so [for the same reason that they did before](https://github.com/cisagov/ansible-role-clamav/pull/70#issuecomment-1821645724).  The issue is that some GH Actions IPs are being blocked by ClamAV.

**Review by jsf9k — APPROVED** [review]
Thank you for the contribution, @desebjohnston!

File: tasks/main.yml
Comment (dav3r) [inline tasks/main.yml]: Minor capitalization quibble:
```suggestion
- name: Install ClamAV from package URL
```

**Review by dav3r — APPROVED** [review]
Looks great, thank you for contributing! 🚀

**Review by dv4harr10 — APPROVED** [review]

**Pull request merged**
